### PR TITLE
Str to path

### PR DESCRIPTION
**Pull request by DanielElisenberg**
# STRING TO PATH

Consistently use pathlib instead of strings to get better type hinting (and to be empathic to a future windows developer).

### Note
One "scary" thing I just found out about:
> If a segment is an absolute path, all previous segments are ignored (like [os.path.join()](https://docs.python.org/3/library/os.path.html#os.path.join))

This means that if you do:
```py
from pathlib import Path
j = Path('job_executor')
j / 'adapter'
# PosixPath('job_executor/adapter')
j / '/adapter'
# PosixPath('/adapter')
```

So if you typo a slash in the extending string it will go directly to the root path. Not ideal imho. But both our unit tests and other validations seem to cover this fairly well so I am not too worried. (Went through the code base and double checked as well)

### CODE REVIEW

File: tests/unit/worker/steps/test_dataset_converter.py [hunk 1]
```diff
@@ -73,88 +73,88 @@ def test_create_list_of_fields_for_partitioned_parquet():
...
             CSV_FILE_READY_FOR_PARQUET_CONVERSION,
             temporality_type="EVENT",
             data_type=data_type
         )
     )
-    parquet_file = pq.read_table(output_parquet_file)
+    parquet_file = pq.read_table(output_parquet_path)
     verify_parquet_file_schema(parquet_file, data_type)
-    assert output_parquet_file.endswith('.parquet')
+    assert str(output_parquet_path).endswith('.parquet')
```
Comment (pawbu) [inline tests/unit/worker/steps/test_dataset_converter.py:86]: Could be:
`output_parquet_path.suffix == '.parquet'`

**Review by pawbu — APPROVED** [review]
Looks good, some minor comments

File: job_executor/worker/steps/dataset_pseudonymizer.py [hunk 2]
```diff
@@ -32,7 +33,9 @@ def _pseudonymize_identifier_only(
     identifier_to_pseudonym = pseudonym_service.pseudonymize(
         list(unique_identifiers), unit_id_type, job_id
     )
-    output_csv_path = input_csv_path.replace('.csv', '_pseudonymized.csv')
+    output_csv_path = Path(
+        str(input_csv_path).replace('.csv', '_pseudonymized.csv')
```
[the remark this answers is not in the document]
Comment (DanielElisenberg) [reply job_executor/worker/steps/dataset_pseudonymizer.py:37]: The file ends up only being named '_pseudonymized.csv', but found another solution that might look prettier than casting the whole path to string 🙌🏻 :
```py
from pathlib import Path

p = Path('/some/dir/my_file.csv')
Path(p.parents[0]) / '_pseudonymized.csv'
# PosixPath('/some/dir/_pseudonymized.csv')
>>> p.split()

p.parent / p.name.replace('.csv', '_pseudonymized.csv')
# PosixPath('/some/dir/my_file_pseudonymized.csv')
```

[the remark this answers is not in the document]
Comment (pawbu) [reply job_executor/worker/steps/dataset_pseudonymizer.py:37]: Yes, this is nice. Also working:
```
>>> Path(p.parents[0]) / f'{p.stem}_pseudonymized.csv'
PosixPath('/some/dir/my_file_pseudonymized.csv')
```

[the remark this answers is not in the document]
Comment (pawbu) [reply job_executor/worker/steps/dataset_pseudonymizer.py:37]: Maybe you solution is clearer as to what we are doing here.

**Review by pawbu — APPROVED** [review]
LGTM

**Pull request merged**